src/viewer.js

```diff
@@ -418,20 +434,71 @@ $.Viewer = function( options ) {
         maxImageCacheCount: this.maxImageCacheCount
     });
 
-    // Create the drawer
-    this.drawer = new $.Drawer({
-        viewer:             this,
-        viewport:           this.viewport,
-        element:            this.canvas,
-        debugGridColor:     this.debugGridColor
-    });
+    //Create the drawer based on selected options
+    if (Object.prototype.hasOwnProperty.call(this.drawerOptions, 'useCanvas') ){
+        $.console.error('useCanvas is deprecated, use the "drawer" option to indicate preferred drawer(s)');
+
+        // for backwards compatibility, use HTMLDrawer if useCanvas is defined an is falsey
+        if (!this.drawerOptions.useCanvas){
+            this.drawer = $.HTMLDrawer;
+        }
+
+        delete this.drawerOptions.useCanvas;
+    }
+    let drawerPriority = Array.isArray(this.drawer) ? this.drawer : [this.drawer];
+    let drawersToTry = drawerPriority.filter(d => ['webgl', 'context2d', 'html'].includes(d) || (d.prototype && d.prototype.isOpenSeadragonDrawer) );
+    if(drawerPriority.length !== drawersToTry.length){
+        $.console.error('An invalid drawer was requested.');
+    }
+    if(drawersToTry.length === 0){
+        drawersToTry = [$.DEFAULT_SETTINGS.drawer].flat(); // ensure it is a list
+        $.console.warn('No valid drawers were selected. Using the default value.');
+    }
+    // extend the drawerOptions object with additional properties to pass to the Drawer implementation
+    this.drawer = null; // TO DO: how to deal with the possibility that none of the requested drawers are supported?
+    for(let i = 0; i < drawersToTry.length; i++){
+        let Drawer = drawersToTry[i];
+        let optsKey = null;
+        // replace text-based option with appropriate constructor
+        if (Drawer === 'context2d'){
+            Drawer = $.Context2dDrawer;
+            optsKey = 'context2d';
+        } else if (Drawer === 'html'){
+            Drawer = $.HTMLDrawer;
+            optsKey = 'html';
+        } else if (Drawer === 'webgl'){
+            Drawer = $.WebGLDrawer;
+            optsKey = 'webgl';
+        } else {
+            optsKey = 'custom';
+        }
+        // if the drawer is supported, create it and break the loop
+        if (Drawer.prototype.isSupported()){
+            this.drawer = new Drawer({
+                viewer:             this,
+                viewport:           this.viewport,
+                element:            this.canvas,
+                debugGridColor:     this.debugGridColor,
+                options:            this.drawerOptions[optsKey],
+            });
+            this.drawerOptions.constructor = Drawer;
+            // TO DO: add an event that indicates which drawer was instantiated?
+            break;
+        }
+        // TO DO: add an event that indicates that the selected drawer could not be created?
```

Comment:
This does seem like it makes sense as an event (though I guess maybe if we just had an event that said which drawer was created, that would cover it).

Reply:
I'm getting rid of these `// TO DOs`` for now. I've added a mechanism to find out the type of drawer that was created (`getType()`) and if none of the requested drawers are supported, it prints an error message to the console and throws an exception. I'm not totally satisfied that this is the best solution, though. Further discussion would be good.

Reply:
Seems like a reasonable way to go for now!